### DISCUSSION TITLE
[0.75/0.76] Android `minSdk` Bump to API 24 announcement

### DISCUSSION THREAD

**Discussion by alanleedev** [react-native-community/discussions-and-proposals#802] [Announcements]
**tl;dr: React Native 0.75 will be the last version to support Android 6.0 (API 23). React Native 0.76 will bump the min SDK version to Android 7.0 (API 24)**

(Last min SDK version bump was in 0.73 (API 23): [past announcement](https://github.com/react-native-community/discussions-and-proposals/discussions/740))

Hi all, 
We're sharing this announcement to inform our community that we're about to bump the Min SDK version for React Native.

The upcoming version of React Native, version 0.75, will be the **last** version to support Android 6.0 (API level 23). The following minor version, 0.76, will bump the Min SDK version to 24.

React Native 0.75 will follow our usual [release support window](https://github.com/reactwg/react-native-releases/blob/main/docs/support.md), meaning that it will be officially supported till the stable release of React Native 0.78.

We're sharing this announcement now to give **enough time** to app developers to properly communicate this change to your users, if necessary.

## Why we're doing this change

Android 6.0 was released on [Sep 29, 2015](https://en.wikipedia.org/wiki/Android_Marshmallow), and will be 10 years old in 2025.

The Play Store statistics from May 2024 shows how Android 6.0 users account for **~1.4%** of the active devices, a number which is going to decrease as time passes:

![android_distribution_may_2024](https://github.com/user-attachments/assets/87bf1276-c720-4d70-93bc-678b96cd57c0)

There are two major factors we consider for bumping the minSdk amonst other things;
 - sufficient test coverage for old OSes
 - developer resources required to maintain the legacy code

As the current testing infrastructure for React Native runs on `minSdk` 24, we do not have ways to discover issues early for API level 23. This is one of our motivations for bumping the `minSdk`.

By dropping support for older versions of Android, we're able to offer you a better React Native experience and simplify our codebase.

## What does this mean for me?

We believe this change won't affect most developers and is part of the natural mobile update cycles.

In the rare circumstance where you're targeting custom hardware or low-end devices, this change means that users of your applications on Android 6 won't be able to receive **updates** of your apps if you update your project beyond React Native 0.75.
In those cases the recommendation is to use React Native 0.75 as long as possible, or work out a plan to update the `minSdkVersion` of your app.

If you have not migrated to the new architecture but plan to do so then we would suggest to test it out ASAP as you will be stuck with 0.75 for a while.
You should report any critical bugs early so it can be fixed while the support window is still open.

**Comment by zoontek** [comment]
@alanleedev Hi 👋

The last time the minSdk was bumped (to 24), the cumulative distribution percentage of supported versions (≥ API 24) was 97.4%, according to your screenshot. Based on the current percentages, if we bump the minSdk to 27, the cumulative distribution percentage of supported versions will be 97.6%.

This could help library authors and allow deletion of legacy code. WDYT?

<img width="1212" height="862" alt="Screenshot 2026-02-24 at 17 24 22" src="https://github.com/user-attachments/assets/4343dfb9-4124-4586-bea6-fa937455795f" />

**Reply by alanleedev** [reply]
Thanks for the suggestion. Will try to have to look into this and gather internal feedback.
